### PR TITLE
add: request sharding, diskless playback; fix: #10, #14

### PR DESCRIPTION
**Pull request by kendfss**
Solves #10 by splitting text into different requests and then combining the output to produce an playable audio stream.
Also tried to help with #14 and #15 by editing the Readme a bit.
Tried to do #5 but I've still not found any docs for the api, or anybody using it with a request that uses fields that this module doesn't. If you don't mind me asking, how did you find out about it? Seems so obscure, lol

### CODE REVIEW

File: htgotts.go [hunk 1]
```diff
@@ -68,46 +73,51 @@ func (speech *Speech) Speak(text string) error {
...
+	}
+	return speech.Handler.PlayBuf(buf)
+}
+
+// Create the folder if does not exists.
 func (speech *Speech) createFolderIfNotExists(folder string) error {
 	dir, err := os.Open(folder)
 	if os.IsNotExist(err) {
-		return os.MkdirAll(folder, 0700)
+		return os.MkdirAll(folder, 0o700)
```
Comment (hegedustibor) [inline htgotts.go:89]: Please, remove this "o" from the permission code.

File: htgotts.go [hunk 2]
```diff
@@ -68,46 +73,51 @@ func (speech *Speech) Speak(text string) error {
...
 		return err
 	}
 
 	f.Close()
 	return nil
 }
 
-func (speech *Speech) generateHashName(name string) string {
-	hash := md5.Sum([]byte(name))
+func (speech *Speech) fnameFromHash(text string) string {
```
Comment (hegedustibor) [inline htgotts.go:120]: Please, if possible, do not use short names for the readable code. My suggestion for example `fileNameFromHash`

**Review by hegedustibor — CHANGES_REQUESTED** [review]
Thank you very much that you send this PR. 🚀 Please check my suggestions for the merge process.

**Review by hegedustibor — APPROVED** [review]

File: htgotts.go (hunk 1 above)
Comment (kendfss) [reply htgotts.go]: In the first line, 0700, the prefix "0" indicates an octal literal, and the digits "700" represent the permission value in octal. This octal value translates to rwx------ in the symbolic representation, which means the owner has read, write, and execute permissions, while others have no permissions.

In the second line, 0o700, the prefix "0o" is an alternative representation introduced in Go 1.13 to make it more explicit that the value is an octal literal. The digits "700" still represent the permission value in octal, and it has the same meaning as in the first line.

I think the change was made by my language server though, not me, so it's probably worth defining project specific settings (assuming you can do so for gopls) if this refactor is to be avoided. I for one think it helps though.

**Comment by kendfss** [comment]
sorry for the delay man. the truth is, I'm pretty sure there's a bug that will cause the algorithm to break for at least chinese and arabic and I wanted to find the time to sit down and properly figure it out but I just cant recover the train of thought I had to notice it. Never having been much of a linguist, I wasn't sure where to start either, lol.
That said, I notice the changes aren't in in the main branch. If you want, to re-open, I can at least implement the requests you made. Otherwise, I could resubmit with them in mind.

**Pull request closed without merging**